main.tf

```diff
@@ -324,3 +326,53 @@ resource "null_resource" "confirm_network_healthy" {
     }
   }
 }
+
+
+resource "ibm_container_addons" "addons" {
+  count             = var.enable_autoscaling && !(var.disable_public_endpoint) ? 1 : 0
+  cluster           = local.cluster_name
+  resource_group_id = var.resource_group_id
+
+  addons {
+    name    = "cluster-autoscaler"
+    version = var.cluster_autoscaler_version
+  }
+
+  lifecycle {
+    ignore_changes = [addons]
+  }
+}
+
+resource "time_sleep" "wait_operators" {
+  depends_on      = [ibm_container_addons.addons]
+  create_duration = "5s"
+}
+
+locals {
+  worker_pool_config = [
+    for worker in var.worker_pools :
+    {
+      name    = worker.pool_name
+      minSize = worker.minSize
+      maxSize = worker.maxSize
+      enabled = worker.enableAutoscaling
+    }
+  ]
```

Comment:
What happens when a consumer specifies more than one worker pool enabled?

Multiple concerns, going from worker_pool's' to a worker pool config. Starting with does the auto scaler add-on support multiple worker configs? Can it auto scale multiple worker pools in a single cluster.

Also how does this module support making the pods associated with the auto scaler run in a specific worker pool? For the consuming modules, for example, a standard example with edge, private and transit worker pools, how do the pods end up on the private worker pool?